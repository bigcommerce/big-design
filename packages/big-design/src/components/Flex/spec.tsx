import { render } from '@testing-library/react';
import React, { createRef } from 'react';
import 'jest-styled-components';

import { Flex, FlexItem } from './index';

test('render flex', () => {
  const { container } = render(<Flex>Flex</Flex>);

  expect(container.firstChild).toMatchSnapshot();
});

test('has display flex', () => {
  const { container } = render(<Flex>Flex</Flex>);

  expect(container.firstChild).toHaveStyle('display: flex');
});

test('has gap properties', () => {
  const { container, rerender } = render(<Flex flexGap="3rem">Flex</Flex>);

  expect(container.firstChild).toHaveStyle('gap: 3rem');

  rerender(
    <Flex flexColumnGap="1rem" flexRowGap="2rem">
      Flex
    </Flex>,
  );

  expect(container.firstChild).toHaveStyle({ 'row-gap': '2rem', 'column-gap': '1rem' });
});

test('forwards styles', () => {
  const { container } = render(
    <Flex className="test" style={{ background: 'red' }}>
      Flex
    </Flex>,
  );

  expect(container.getElementsByClassName('test')).toHaveLength(1);
  expect(container.firstChild).toHaveStyle('background: red');
});

test('Flex Item forwards styles', () => {
  const { container } = render(
    <FlexItem className="test" style={{ background: 'red' }}>
      Flex
    </FlexItem>,
  );

  expect(container.getElementsByClassName('test')).toHaveLength(1);
  expect(container.firstChild).toHaveStyle('background: red');
});

test('rendering as another element retains inherited props and styles', () => {
  const { getByTestId } = render(<Flex as="section" data-testid="flex" margin="medium" />);

  const flex = getByTestId('flex');

  expect(flex.tagName).toBe('SECTION');
  expect(flex).toHaveStyle(`margin: 1rem`);
});

test('Flex Item should handle falsy values (0)', () => {
  const { getByTestId } = render(<FlexItem data-testid="flex" flexShrink={0} />);

  const flex = getByTestId('flex');

  expect(flex).toHaveStyle('flex-shrink: 0');
});

test('Flex forwards ref', () => {
  const ref = createRef<HTMLDivElement>();

  const { container } = render(<Flex ref={ref}>Hello</Flex>);
  const div = container.querySelector('div');

  expect(div).toBe(ref.current);
});

test('FlexItem forwards ref', () => {
  const ref = createRef<HTMLDivElement>();

  const { container } = render(<FlexItem ref={ref}>Hello</FlexItem>);
  const div = container.querySelector('div');

  expect(div).toBe(ref.current);
});
