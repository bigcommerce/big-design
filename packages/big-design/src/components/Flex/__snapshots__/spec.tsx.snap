// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`render flex 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (min-width:0px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:720px) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div
  class="c0 c1"
>
  Flex
</div>
`;
